core/src/chain/pair.rs

```diff
@@ -12,37 +14,27 @@ pub struct Pair {
 }
 
 impl Pair {
-    /// build a new Pair from a chain and entry
-    ///
-    /// Header is generated automatically
-    ///
-    /// a Pair is immutable, but the chain is mutable if chain.push() is used.
-    ///
-    /// this means that if two Pairs X and Y are generated for chain C then Pair X is pushed onto
-    /// C to create chain C' (containing X), then Pair Y is no longer valid as the headers would
-    /// need to include X. Pair Y can be regenerated with the same parameters as Y' and will be
-    /// now be valid, the new Y' will include correct headers pointing to X.
-    ///
-    /// # Panics
-    ///
-    /// Panics if entry is somehow invalid
-    ///
-    /// @see chain::entry::Entry
-    /// @see chain::header::Header
-    pub fn new(chain: &Chain, entry: &Entry) -> Pair {
-        let header = Header::new(chain, entry);
+    /// Reconstruct Pair from Header stored in a HashTable
+    pub fn new_from_header(table: &ActorRef<Protocol>, header: &Header) -> Option<Self> {
```

Comment:
Same as above. I think `from_header` is enough. This is a very common naming scheme for constructors.

Reply:
fixed